### PR TITLE
Fix generated since information

### PR DESCRIPTION
**Pull request by parlough**
In my [previous change](https://github.com/dart-lang/linter/commit/f594bec4098a5cb12995d85cc5497b18ce223d58), looks like I accidentally commited "linter" instead of "sdk" for the repo name, breaking new since information. You can see this for example on [`library_annotations`](https://dart-lang.github.io/linter/lints/library_annotations.html) which says it's still unreleased.

This also adds handling for when the SDK DEPS file specifies a commit not included in a linter tag (such as currently on 3.0.0).

Tested by generating the docs with accurate release information.

### CODE REVIEW

**Review by pq — CHANGES_REQUESTED** [review]

File: tool/since.dart [hunk 1]
```diff
@@ -5,31 +5,29 @@
...
-  // }
-
-  // Uncomment to (re)generate since/dart_sdk.yaml contents.
-  // for (var info in sinceInfo.entries) {
-  //   var sinceSdk = info.value.sinceDartSdk;
-  //   if (sinceSdk != null) {
-  //     print('${info.key}: $sinceSdk}');
-  //   }
-  // }
+  for (var MapEntry(key: lintName, value: sinceInfo) in sinceInfo.entries) {
```
[the remark this answers is not in the document]
Comment (parlough) [reply tool/since.dart:18]: 👍 Let me add some use of `package:args` here.

[the remark this answers is not in the document]
Comment (pq) [reply tool/since.dart:18]: Awesome. Thanks!

**Review by pq — APPROVED** [review]
Thanks!

**Comment by parlough** [comment]
Thanks @pq! Worked well :)

**Comment by pq** [comment]
Awesome. Thank **YOU**!

**Pull request merged**